argo-cd-apps/base/spi.yaml

```diff
@@ -41,7 +41,7 @@ spec:
             duration: 15s # the amount to back off. Default unit is seconds, but could also be a duration (e.g. "2m", "1h")
 ---
 apiVersion: argoproj.io/v1alpha1
-kind: ApplicationSet
+kind: Application
 metadata:
   name: spi-vault
 spec:
```

Comment:
generators and template should be removed

Reply:
should be similar to https://github.com/redhat-appstudio/infra-deployments/blob/pre-kcp/argo-cd-apps/base/spi.yaml

also https://github.com/redhat-appstudio/infra-deployments/blob/main/argo-cd-apps/overlays/development/repo-overlay.yaml#L34-L50 should be updated